### PR TITLE
AR-1681 Create AlertBanner component 

### PR DESCRIPTION
**Pull request by Jephuff**
Creates an AlertBanner component for inline alerts
https://www.sketch.com/s/aa84f7e1-c1aa-4f8c-a252-b62955de353a/a/ew5w0x#Inspector
image

### CODE REVIEW

File: src/AlertBanner/AlertBanner.tsx [hunk 1]
```diff
@@ -0,0 +1,102 @@
...
+      case "warn":
+        return { color: colors.orange, Icon: IconWarningSolid };
+      case "error":
+        return { color: colors.red, Icon: IconErrorSolid };
+      case "success":
+        return { color: colors.green, Icon: IconSuccessSolid };
+      default:
+        assertUnreachable(type);
+    }
+  }, [type]);
```
Comment (justinanastos) [inline src/AlertBanner/AlertBanner.tsx:54]: FYI: I originally didn't like this pattern, but it's grown on me as I've seen it more 👍

File: src/AlertBanner/AlertBanner.tsx [hunk 2]
```diff
@@ -0,0 +1,102 @@
...
+  type: "info" | "warn" | "error" | "success";
+}
+
+export const AlertBanner: React.FC<AlertBannerProps> = ({
+  children,
+  theme = "light",
+  type,
+  ...otherProps
+}) => {
+  const { Icon, color: colorTemp } = useMemo(() => {
```
Comment (justinanastos) [inline src/AlertBanner/AlertBanner.tsx:41]: NBD: I don't feel strongly about this, but the misdirection for `color: colorTemp` might not be necessary.

File: src/AlertBanner/AlertBanner.tsx [hunk 3]
```diff
@@ -0,0 +1,102 @@
...
+      }}
+    >
+      <Icon
+        css={{
+          width: 20,
+          height: 20,
+          color: colorTemp.base,
+          marginRight: 13,
+          "& .inner": theme === "dark" &&
+            type !== "warn" && { fill: colors.white },
```
Comment (justinanastos) [inline src/AlertBanner/AlertBanner.tsx:90]: 😭

**Comment by justinanastos** [comment]
One thing unrelated to the review; I want to investigate the "team review" features in Chromatic; I think that we should try to get away from images int he PR description because there's no way to know if those are up to date or not.

**Comment by apollo-bot2** [comment]
:rocket: PR was released in [`v7.2.0`](https://github.com/apollographql/space-kit/releases/tag/v7.2.0) :rocket:

File: src/AlertBanner/AlertBanner.tsx (hunk 3 above)
Comment (jglovier) [reply src/AlertBanner/AlertBanner.tsx:90]: Yeah, the more I've thought about this the more it seems kind of ridiculous even from a design perspective. Maybe we can change the color value of the triangle to something that makes it work well with just white in either scenario. I kinda hate having that inconsistency too.

**Comment by jglovier** [comment]
@Jephuff awesome work!! Thanks for getting this in.
Rick-Ross-Boom

File: src/AlertBanner/AlertBanner.story.mdx [hunk 4]
```diff
@@ -0,0 +1,91 @@
...
+  <Story name="success light">
+    <AlertBanner type="success">
+      This is an example of an inline error message. It should be rather short,
+      but can have multi-lines if needed. Learn more
+    </AlertBanner>
+  </Story>
+  <Story name="success dark">
+    <AlertBanner type="success" theme="dark">
+      This is an example of an inline error message. It should be rather short,
+      but can have multi-lines if needed. Learn more
```
Comment (jglovier) [inline src/AlertBanner/AlertBanner.story.mdx:82]: The `Learn more` in these should really be linked up (or) to demonstrate how to use links within the banner. Or if you were thinking that the whole banner would be a target, that works too, but the underline on the `Learn more` text is still a helpful visual clue.

Comment (Jephuff) [reply src/AlertBanner/AlertBanner.story.mdx:82]: oops, I meant to remove the learn nows to simplify the story :D
I'll make another PR to clean this up and remove the shadow 👍

**Pull request merged**